build_tools/python/cmake_builder/rules.py

```diff
@@ -0,0 +1,149 @@
+## Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Helpers that build CMake rules."""
+
+from typing import List, Optional, Sequence
+
+INDENT_SPACES = " " * 2
+
+
+def _get_string_list(values: List[str], quote: bool = True) -> List[str]:
+  if quote:
+    values = [f"\"{value}\"" for value in values]
+  return values
+
+
+def _get_block_body(body: List[str]) -> List[str]:
+  return [INDENT_SPACES + line for line in body]
+
+
+def _get_string_arg_block(keyword: str,
+                          value: Optional[str],
+                          quote: bool = True) -> List[str]:
+  if value is None:
+    return []
+  if quote:
+    value = f"\"{value}\""
+  return [keyword] + _get_block_body([value])
+
+
+def _get_string_list_arg_block(keyword: str,
+                               values: List[str],
+                               quote: bool = True) -> List[str]:
+  body = _get_string_list(values, quote)
+  if len(body) == 0:
```

Comment:
Seems like we can do this check immediately on values before generating the body?